### PR TITLE
Add advanced vision tutorial

### PR DESCRIPTION
**Pull request by tkclam**
### Description
Add tutorial for connectome-constrained visual model + multi-fly simulation. Implemented `HybridTurningFly` and `RealisticVisionFly` as alternatives to `HybridTurningController` and `RealisticVisionController` (inherit from `Fly` instead of `Simulation` to allow having different types of flies in one simulation).

### Types of changes

- [ ] Bugfix
- [ ] New feature
- [x] Refactor / Code style update (no logical changes)
- [ ] Build / CI changes
- [ ] Documentation Update
- [x] Other (explain)

### Does this address any currently open issues?
#157

### CODE REVIEW

**Comment by sibocw** [comment]
Thank you! I will work on this soon.

File: flygym/examples/vision/realistic_vision_fly.py
Comment (sibocw) [inline flygym/examples/vision/realistic_vision_fly.py:1]: Maybe `RealisticVisionFly` can be moved to `realistic_vision.py`?

File: flygym/examples/vision/vision_network.py
Comment (sibocw) [inline flygym/examples/vision/vision_network.py:1]: thanks for fixing this!

File: flygym/simulation.py [hunk 1]
```diff
@@ -321,6 +321,10 @@ def close(self) -> None:
                 camera.output_path.parent.mkdir(parents=True, exist_ok=True)
                 camera.save_video(camera.output_path)
 
+        for fly in self.flies:
+            if hasattr(fly, "close"):
+                fly.close()
+
```
Comment (sibocw) [inline flygym/simulation.py:327]: Why would some flies not have the close method? Should we add an empty `.close()` to the fly base class then?

File: notebooks/connectome_vision_model.ipynb
Comment (sibocw) [inline notebooks/connectome_vision_model.ipynb:1]: Can you add a paragraph explaining the difference between HybridTurningController and HybridTurningFly?

**Review by sibocw — CHANGES_REQUESTED** [review]
Hi Tommy, can you check the following comments, and add a paragraph in the tutorial to explain the difference between the HybridTurningController and the HybridTurningFly please?

File: notebooks/connectome_vision_model.ipynb
Comment (tkclam) [reply notebooks/connectome_vision_model.ipynb:1]: Shall we replace `HybridTurningController` with `HybridTurningFly` and remove `HybridTurningFly` to reduce code duplications?

Comment (sibocw) [reply notebooks/connectome_vision_model.ipynb:1]: That sounds good, but probably a lot of things need to change... i'm not sure if it's worth redoing everything again...

**Review by sibocw — APPROVED** [review]

**Pull request merged**